packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -219,33 +226,133 @@ class _ProductPageState extends State<NewProductPage> {
           top: 4.0,
           bottom: 20.0,
         ),
-        insets: const EdgeInsets.all(12.0),
-        child: Column(children: <Widget>[
-          Align(
+        insets: EdgeInsets.zero, // Zero padding for the card content.
+        clipBehavior: Clip.antiAliasWithSaveLayer,
+        child: Column(
+          children: <Widget>[
+            getProductMatchHeader(context),
+            Container(
+              padding: const EdgeInsets.symmetric(horizontal: 12.0),
+              child: Column(children: [
+                getProductTitleTile(context),
+                for (final Attribute attribute in scoreAttributes)
+                  ScoreAttributeCard(
+                      attribute: attribute, iconHeight: iconHeight),
+                attributesContainer,
+              ]),
+            ),
+          ],
+        ),
+      ),
+    );
+    return ListView(children: listItems);
+  }
+
+  Widget getProductMatchHeader(BuildContext context) {
+    // NOTE: This is temporary and will be updated once the feature is supported
+    // by the server.
+    return Container(
+      color: Colors.red,
+      padding: EdgeInsets.zero,
+      child: Container(
+        alignment: Alignment.topLeft,
+        padding: const EdgeInsets.symmetric(vertical: 8),
+        child: Center(child: Text('Very poor Match', style: Theme.of(context).textTheme.subtitle1)),
+      ),
+    );
+  }
+
+  Widget getProductTitleTile(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final ThemeData themeData = Theme.of(context);
+    return Align(
+      alignment: Alignment.topLeft,
+      child: ListTile(
+        contentPadding: EdgeInsets.zero,
+        title: Text(
+          _getProductName(appLocalizations),
+          style: themeData.textTheme.headline4,
+        ),
+        subtitle: Text(_product.brands ?? appLocalizations.unknownBrand),
+        trailing: Text(
+          _product.quantity ?? '',
+          style: themeData.textTheme.headline3,
+        ),
+      ),
+    );
+  }
+
+  Widget getAttributeGroupWidget(
```

Comment:
`getAttributeGroupWidget()` is a bit generic. How about `buildAttributeGroupView()` or `...ListTile()` or `..Row()`?

Reply:
`_buildAttributeGroupContainer` ?

Reply:
"Container" sounds to me like it takes a `child` to wrap. How about just `_buildAttributeGroup`?

Reply:
Sure